web/src/app/profiles-list/page.tsx

```diff
@@ -44,13 +73,54 @@ export default async function ProfilesList() {
                     <Input
                         className="rounded-r-none"
                         placeholder="Busque por um perfil"
+                        value={search}
+                        onChange={updateSearch}
                     />
-                    <Button className="bg-indigo-950 rounded-sm rounded-l-none">
+                    <Button
+                        onClick={handleSearch}
+                        className="bg-indigo-950 rounded-sm rounded-l-none"
+                    >
                         <Search />
                     </Button>
                 </div>
             </div>
-            <ProfileCard />
+            {profiles
+                .filter((profile) => {
+                    if (!filter) {
+                        return profile;
+                    }
+                    if (
+                        profile.name
+                            .toLowerCase()
+                            .includes(filter.toLowerCase())
+                    ) {
+                        return profile;
+                    }
+                    if (
+                        profile.email
+                            .toLowerCase()
+                            .includes(filter.toLowerCase())
+                    ) {
+                        return profile;
+                    }
+                    if (
+                        profile.level
+                            .toLowerCase()
+                            .includes(filter.toLowerCase())
+                    ) {
+                        return profile;
+                    }
+                    if (
+                        profile.occupation
+                            .toLowerCase()
+                            .includes(filter.toLowerCase())
+                    ) {
+                        return profile;
+                    }
```

Comment:
```suggestion
if (!filter) {
    return true;
}

const formattedFilter = filter.toLowerCase();
const searchableKeys = ['name', 'email', 'level', 'occupation'];

const profileMatchFilter = searchableKeys.find((key) => {
    return profile[key].toLowerCase().includes(formattedFilter);
});

return profileMatchFilter;
```

Essa mudança deixa o código menos repetitivo.
`searchableKeys` define um array de chaves as quais a gente vai usar pra encontrar resultados.
`profileMatchFilter` vai ser true caso alguma das `searchableKeys` bata com o valor do filtro.

Reply:
Feito